source/Ship.cpp

```diff
@@ -721,6 +721,10 @@ void Ship::FinishLoading(bool isNewInstance)
 		warning += "Defaulting " + string(attributes.Get("drag") ? "invalid" : "missing") + " \"drag\" attribute to 100.0\n";
 		attributes.Set("drag", 100.);
 	}
+
+	attraction = CalculateAttraction();
+	deterrence = CalculateDeterrence();
```

Comment:
The Calculate functions already modify the data members. Either make them not modify them or remove the assignments here.